experimental/ComplexReflectionGroups/src/ComplexReflectionGroupType.jl

```diff
@@ -0,0 +1,680 @@
+            if m == 1 && p == 1 # symmetric group case
+                append!(codegrees, convert(Vector{ZZRingElem}, collect(0:n-2) ))
+            elseif n == 1 # cyclic group case
+                append!(codegrees, convert(Vector{ZZRingElem}, [0]) )
+            else
+                if p != m
+                    append!(codegrees, [ZZ(m)*i for i=0:n-1])
+                else
+                    append!(codegrees, sort([[ZZ(m)*i for i=0:n-2]; [ZZ(n-1)*ZZ(m)-ZZ(n)]]))
+                end
+            end
+        end 
+    end
+
+    return codegrees
+
+end
+
+function exponents(G::ComplexReflectionGroupType)
+
+    # See Lehrer & Taylor (2009), p 51
+    deg = degrees(G)
+    return [ d-1 for d in deg ]
+
+end 
+
+function coexponents(G::ComplexReflectionGroupType)
+
+    # See Lehrer & Taylor (2009), p 205
+    codeg = codegrees(G)
+    return [ d+1 for d in codeg ]
+
+end 
+
+function num_hyperplanes(G::ComplexReflectionGroupType)
+
+    # See Geck & Malle (2003), Section 1.7
+
+    coexp = coexponents(G)
+    return sum(coexp)
+
+end
+
+function is_pseudo_real(G::ComplexReflectionGroupType)
+
+    # Grabbed from Magma
+    excpreal = [ 12, 13, 22, 23, 24, 27, 28, 29, 30, 31, 33, 34, 35, 36, 37 ]
+
+    for t in G.type
+        if isa(t, Int) == 1
+            if !(t in excpreal)
+                return false
+            end
+        else
+            # See Lehrer & Taylor (2009). Exercise 2.15 (p 38)
+            m = t[1]
+            p = t[2]
+            n = t[3]
+            if !(m == p || ZZ(m) == 2*ZZ(p))
+                return false
+            end
+        end
+    end
+
+    return true
+end
+
+function is_real(G::ComplexReflectionGroupType)
+
+    excreal = [ 23, 28, 30, 35, 36, 37 ]
+
+    for t in G.type
+        if isa(t, Int)
+            if !(t in excreal)
+                return false
+            end
+        else
+            m = t[1]
+            p = t[2]
+            n = t[3]
+            # Only real groups are symmetric, dihedral, type B, and type D
+            if !( (m == 1 && p == 1) || (m == p && n == 2) || (m == 2 && p == 1) || ( m == 2 && p == 2) )
+                return false
+            end
+        end
+    end
+
+    return true
+
+end
+
+function is_coxeter_group(G::ComplexReflectionGroupType)
+    return is_real(G)
+end
+
+function is_rational(G::ComplexReflectionGroupType)
+
+    excrat = [ 28, 35, 36, 37 ]
+
+    for t in G.type
+        if isa(t, Int)
+            if !(t in excrat)
+                return false
+            end
+        else
+            m = t[1]
+            p = t[2]
+            n = t[3]
+            # Only rational groups are symmetric, dihedral order 12, type B, and type D
+            if !( (m == 1 && p == 1) || (m == 6 && p == 6 && n == 2) || (m == 2 && p == 1) || ( m == 2 && p == 2) )
+                return false
+            end
+        end
+    end
+
+    return true
+end
+
+function is_weyl_group(G::ComplexReflectionGroupType)
+    return is_rational(G)
+end
```

Comment:
```suggestion
@alias is_weyl_group is_rational
```
(as above for `is_coxeter_group`)

Reply:
I would disagree as there are other dispatches for `is_rational` (like `is_rational(x::QQBarFieldElem)`) that would not make sense for `is_weyl_group`. `@alias` should only be used if all methods make sense for both names.